[docs] Restructure Going beyond SQL (#22080)
yugabyte/yugabyte-db: docs/content/preview/explore/going-beyond-sql/cluster-aware-drivers.md

## External load balancer

Typically when applications have to connect to a multi-node system, they typically use a load balancer which would accept the multiple connections from the client and distribute them across the different nodes in the system underneath it. This would also be the case for an application connecting to a YugabyteDB cluster using a standard PostgreSQL driver.

![Using an external load balancer](/images/explore/scalability/node-addition-loadbalancer.png)

Although the load balancer hides the cluster from the application, neither the load balancer nor application will have any idea of the topology of the cluster and every time a node is added to the cluster, the load balancer config has to be updated. Another hidden detail is that the load balancer adds a hop for the application to talk to the cluster.

Smart drivers play a crucial role in simplifying the process of connecting to and communicating with a database cluster without the need for an external load balancer.

## Cluster information

> When an application connects to a cluster using the smart driver, the driver connects any one node in the connection string. Then it fetches information about all the nodes in the universe using the `yb_servers()` function. This list is refreshed at regular intervals (default: 5 minutes). Subsequently, the driver connects to the least-loaded node before returning the connection to the application.

Reviewer: ```suggestion
When an application connects to a cluster using the smart driver, the driver makes an initial connection to a node specified in the connection string. The driver then fetches information about all the nodes in the universe using the `yb_servers()` function. This list is refreshed at regular intervals (the default is 5 minutes). Subsequently, the driver connects to the least-loaded node before returning the connection to the application.
```

## Load balancing

> Now that the driver has fetched information about all the nodes in the cluster, if load balancing is enabled via the connection string option `load_balance=true`, the driver distributes new connection requests made by the application across the different nodes in the cluster. As the application directly connects to the node there is no extra hop here.

Reviewer: ```suggestion
Now that the driver has fetched information about all the nodes in the cluster, if load balancing is enabled via the connection string option `load_balance=true`, the driver distributes new connection requests made by the application across the different nodes in the cluster. In addition, because the application connects directly to the node, there is no extra hop.
```
